torchaudio/functional/filtering.py

```diff
@@ -887,11 +870,43 @@ def lfilter(
 
     output = padded_output_waveform[:, n_order - 1:]
 
-    if clamp:
-        output = torch.clamp(output, min=-1.0, max=1.0)
-
     # unpack batch
     output = output.reshape(shape[:-1] + output.shape[-1:])
+    return output
+
+try:
+    _lfilter = torch.ops.torchaudio._lfilter
+except RuntimeError as err:
+    assert str(err) == 'No such operator torchaudio::_lfilter'
+    _lfilter = _lfilter_core
+
+
+def lfilter(
+    waveform: Tensor,
+    a_coeffs: Tensor,
+    b_coeffs: Tensor,
+    clamp: bool = True,
+) -> Tensor:
+    r"""Perform an IIR filter by evaluating difference equation.
+
+    Args:
+        waveform (Tensor): audio waveform of dimension of ``(..., time)``.  Must be normalized to -1 to 1.
+        a_coeffs (Tensor): denominator coefficients of difference equation of dimension of ``(n_order + 1)``.
+                                Lower delays coefficients are first, e.g. ``[a0, a1, a2, ...]``.
+                                Must be same size as b_coeffs (pad with 0's as necessary).
+        b_coeffs (Tensor): numerator coefficients of difference equation of dimension of ``(n_order + 1)``.
+                                 Lower delays coefficients are first, e.g. ``[b0, b1, b2, ...]``.
+                                 Must be same size as a_coeffs (pad with 0's as necessary).
+        clamp (bool, optional): If ``True``, clamp the output signal to be in the range [-1, 1] (Default: ``True``)
+
+    Returns:
+        Tensor: Waveform with dimension of ``(..., time)``.
+    """
+    # pack batch
+    output = _lfilter(waveform, a_coeffs, b_coeffs)
+
+    if clamp:
+        output = torch.clamp(output, min=-1.0, max=1.0)
```

Comment:
May I suggest we move ```output = output.reshape(shape[:-1] + output.shape[-1:])``` here as well and remove it from _lfilter_core above? That way it'll be applied to both the Python and C++ version. The reason the CI system didn't catch this is because there are no multidimensional tests for lfilter yet.

Reply:
In that case, I would also suggest we move the packing part (L833 ~ L834 in `filtering.py` and L93 ~ L94 in `lfilter.cpp`) here just before calling `_filter`. In this way the required input and output of `_filter` will always be a matrix with shape `(n_channel, times)`.

Reply:
Let's go with this, if you don't mind.